packages/components/src/components/hds/application-state/index.ts

```diff
@@ -22,7 +31,30 @@ export interface HdsApplicationStateSignature {
   Element: HTMLDivElement;
 }
 
-const HdsApplicationStateComponent =
-  TemplateOnlyComponent<HdsApplicationStateSignature>();
+export default class HdsApplicationStateComponent extends Component<HdsApplicationStateSignature> {
+  constructor(owner: unknown, args: HdsApplicationStateSignature['Args']) {
+    super(owner, args);
+
+    const validAlignValues: HdsApplicationStateAligns[] = Object.values(
+      HdsApplicationStateAlignValues
+    );
 
-export default HdsApplicationStateComponent;
+    if (
+      this.args.align != null &&
+      !validAlignValues.includes(this.args.align)
+    ) {
+      throw new Error(`Invalid align value: ${this.args.align}`);
```

Comment:
We don't throw errors, we use `assert()` (see how it's done in other components)

Reply:
And most often than not we do it in the getter itself, see how it's done here for example: https://github.com/hashicorp/design-system/blob/main/packages/components/src/components/hds/accordion/index.ts#L44-L53

(no need to have a constructor for this)

Reply:
Absolutely meant to implement this as an `assert`. Thanks for catching this.